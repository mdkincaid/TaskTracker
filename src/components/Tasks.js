// react
import { useState } from 'react';


import Task from './Task';

const Tasks = ({ tasks, onDelete, onToggle, onEdit }) => {
	const [editing, setEditing] = useState('');

	const setEditStatus = (id) => {
		if (editing === id) {
			setEditing('');
		} else {
			setEditing(id);
		}
	};

	return (
		<>
			{tasks.map((task) => (
				<Task
					key={task.id}
					task={task}
					onDelete={onDelete}
					onToggle={onToggle}
					onEdit={onEdit}
					editing={editing}
					setEditing={setEditStatus}
				/>
			))}
		</>
	);
};

export default Tasks;
