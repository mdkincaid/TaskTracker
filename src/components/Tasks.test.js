// react
import React from 'react';

// third-party
import { shallow } from 'enzyme';

// project
import Tasks from './Tasks';

const tasks = [
	{
		text: 'Test Task',
		date: '2021-06-01T20:30:00.000Z',
		reminder: false,
		id: 14,
	},
	{
		text: 'Food Shopping',
		date: '2021-06-17T01:30:00.000Z',
		reminder: false,
		id: 15,
	},
	{
		text: 'Test',
		date: '2021-06-10T04:00:00.000Z',
		reminder: false,
		id: 16,
	},
];

describe('Tasks', () => {
	it('should render the Tasks Component correctly', () => {
		// arrange / act
		let wrapped = shallow(<Tasks tasks={tasks}></Tasks>);

		// assert
		expect(wrapped).toMatchSnapshot();
	});

	it('should render 3 tasks when 3 tasks are supplied', () => {
		// arrange / act
		let wrapped = shallow(<Tasks tasks={tasks}></Tasks>);
		const expectedCount = tasks.length;

		// assert
		expect(wrapped.find('Task').length).toEqual(expectedCount);
	});

	it('should clear the value of editing when the task id matches editing', () => {
		// TODO(): figure out how to test setEditStatus
	});
});
