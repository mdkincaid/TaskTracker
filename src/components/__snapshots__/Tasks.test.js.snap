// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tasks should render the Tasks Component correctly 1`] = `
<Fragment>
  <Task
    editing=""
    key="14"
    setEditing={[Function]}
    task={
      Object {
        "date": "2021-06-01T20:30:00.000Z",
        "id": 14,
        "reminder": false,
        "text": "Test Task",
      }
    }
  />
  <Task
    editing=""
    key="15"
    setEditing={[Function]}
    task={
      Object {
        "date": "2021-06-17T01:30:00.000Z",
        "id": 15,
        "reminder": false,
        "text": "Food Shopping",
      }
    }
  />
  <Task
    editing=""
    key="16"
    setEditing={[Function]}
    task={
      Object {
        "date": "2021-06-10T04:00:00.000Z",
        "id": 16,
        "reminder": false,
        "text": "Test",
      }
    }
  />
</Fragment>
`;
